ecr/base.go

```diff
@@ -53,14 +53,26 @@ type ecrAPI interface {
 
 // getImageByDescriptor retrieves an image from ECR for a given OCI descriptor.
 func (b *ecrBase) getImageByDescriptor(ctx context.Context, desc ocispec.Descriptor) (*ecr.Image, error) {
+	// Only valid descriptors will be queried for.
+	if err := desc.Digest.Validate(); err != nil {
+		return nil, err
+	}
+
+	ident := &ecr.ImageIdentifier{ImageDigest: aws.String(desc.Digest.String())}
+	// Use stronger query when digest matches the reference.
+	if b.ecrSpec.Spec().Digest() == desc.Digest {
+		if tag, _ := b.ecrSpec.TagDigest(); tag != "" {
+			ident.ImageTag = aws.String(tag)
+		}
+	}
```

Comment:
The benefit here being: if the caller specifies both the tag and digest, returning a manifest that corresponds to the digest but not the tag is unexpected?

Reply:
Yes, the query should be as specific as the user has informed the call to be. In this case, if the digest that was given the parsed resource matches then it should be included in the `BatchGetImage` query. If the digest and tag *no longer match* then the result will be that there is no matching image. It

Reply:
That makes sense, mostly I'd like to see that spelled out a bit more clearly in a comment.

The only concern I have is a race around when tags are set on a particular digest; this stricter comparison turns that race into a failure.  I agree with you though that I think this behavior likely matches the request.

Reply:
Gotcha - I've added a lengthy explanation here in case folks come looking to find why errors might be reported where they weren't before. I don't expect that users are typically specifying *both* the tag and a digest (ex: `ecr.aws/arn:aws:ecr:us-west-2:111111111111:repository/example-repo:tag-name@sha256:3df886cef76fd8027dd6274d91ce32877311dc9de7dacedb14670c3dcc840813`).

Please give the added comment a read and let me know if there's changes that could help it be more clear or concise. Thanks!